conan/tools/cmake/cmake.py

```diff
@@ -304,11 +309,23 @@ def ctest(self, cli_args=None, env="", stdout=None, stderr=None):
         if njobs:
             arg_list.append(f"--parallel {njobs}")
 
+        arg_list.extend(cli_args or [])
+
         verbosity = self._conanfile.conf.get("tools.build:verbosity", choices=("quiet", "verbose"))
         if verbosity:
+            # https://cmake.org/cmake/help/latest/manual/ctest.1.html
+            # Options such as --verbose, --extra-verbose, and --debug are
+            # ignored if --quiet is specified.
+            arg_list = [a for a in arg_list if a not in ("--quiet", "--verbose")]
             arg_list.append(f"--{verbosity}")
 
-        arg_list.extend(cli_args or [])
+        extra_args = self._conanfile.conf.get("tools.cmake:ctest_args", check_type=list)
+        if extra_args:
+            # Remove --quiet if incompatible with new args, as latest arg doesn't win in ctest
+            if any(a in ("--debug", "--verbose", "--extra-verbose") for a in extra_args):
+                arg_list = [a for a in arg_list if a != "--quiet"]
+            arg_list.extend(extra_args)
```

Comment:
I think we can end up with duplicated arguments if they are declared both in the `ctest_args` and `cli_args`? I guess it's fine but maybe we want to filter them?